ddtrace/contrib/openai/patch.py

```diff
@@ -0,0 +1,540 @@
+                    attrs={
+                        "messages": messages,
+                        "completion": completions,
+                    },
+                )
+
+
+class _EmbeddingHook(_EndpointHook):
+    def handle_request(self, integration, span, args, kwargs):
+        for kw_attr in ["model", "input", "user"]:
+            if kw_attr in kwargs:
+                if kw_attr == "input" and isinstance(kwargs["input"], list):
+                    for idx, inp in enumerate(kwargs["input"]):
+                        span.set_tag_str("request.input.%d" % idx, integration.trunc(inp))
+                span.set_tag("request.%s" % kw_attr, kwargs[kw_attr])
+
+        resp, error = yield
+
+        if resp:
+            if "data" in resp:
+                span.set_tag("response.data.num-embeddings", len(resp["data"]))
+                span.set_tag("response.data.embedding-length", len(resp["data"][0]["embedding"]))
+            for kw_attr in ["object", "usage"]:
+                if kw_attr in kwargs:
+                    span.set_tag("response.%s" % kw_attr, kwargs[kw_attr])
+            integration.record_usage(span, resp.get("usage"))
+
+
+def _patched_convert(integration):
+    @trace_utils.with_traced_module
+    def patched_convert(openai, pin, func, instance, args, kwargs):
+        """Patch convert captures header information in the openai response"""
+        for val in args:
+            if isinstance(val, openai.openai_response.OpenAIResponse):
+                span = pin.tracer.current_span()
+                if not span:
+                    return func(*args, **kwargs)
+                val = val._headers
+                if val.get("openai-organization"):
+                    org_name = val.get("openai-organization")
+                    span.set_tag("organization.name", org_name)
+
+                # Gauge total rate limit
+                if val.get("x-ratelimit-limit-requests"):
+                    v = val.get("x-ratelimit-limit-requests")
+                    integration.metric(span, "gauge", "ratelimit.requests", v)
+                if val.get("x-ratelimit-limit-tokens"):
+                    v = val.get("x-ratelimit-limit-tokens")
+                    integration.metric(span, "gauge", "ratelimit.tokens", v)
+
+                # Gauge and set span info for remaining requests and tokens
+                if val.get("x-ratelimit-remaining-requests"):
+                    v = val.get("x-ratelimit-remaining-requests")
+                    integration.metric(span, "gauge", "ratelimit.remaining.requests", v)
+                    span.set_tag("organization.ratelimit.requests.remaining", v)
+                if val.get("x-ratelimit-remaining-tokens"):
+                    v = val.get("x-ratelimit-remaining-tokens")
+                    integration.metric(span, "gauge", "ratelimit.remaining.tokens", v)
+                    span.set_tag("organization.ratelimit.tokens.remaining", v)
+        return func(*args, **kwargs)
```

Comment:
```suggestion
        span = pin.tracer.current_span()
        if not span:
            return func(*args, **kwargs)

        for val in args:
            if not isinstance(val, openai.openai_response.OpenAIResponse):
                continue
                
            val = val._headers
            if val.get("openai-organization"):
                org_name = val.get("openai-organization")
                span.set_tag("organization.name", org_name)

            # Gauge total rate limit
            if val.get("x-ratelimit-limit-requests"):
                v = val.get("x-ratelimit-limit-requests")
                integration.metric(span, "gauge", "ratelimit.requests", v)
            if val.get("x-ratelimit-limit-tokens"):
                v = val.get("x-ratelimit-limit-tokens")
                integration.metric(span, "gauge", "ratelimit.tokens", v)

            # Gauge and set span info for remaining requests and tokens
            if val.get("x-ratelimit-remaining-requests"):
                v = val.get("x-ratelimit-remaining-requests")
                integration.metric(span, "gauge", "ratelimit.remaining.requests", v)
                span.set_tag("organization.ratelimit.requests.remaining", v)
            if val.get("x-ratelimit-remaining-tokens"):
                v = val.get("x-ratelimit-remaining-tokens")
                integration.metric(span, "gauge", "ratelimit.remaining.tokens", v)
                span.set_tag("organization.ratelimit.tokens.remaining", v)
        return func(*args, **kwargs)
```

Reply:
alternative option. checking for active span first will short circuit the function if we aren't going to do anything anyways. moving the isinstance check to a not with a continue removed one level of indentation.